types/types.go

```diff
@@ -144,6 +145,10 @@ type ImageDestination interface {
 	// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination.
 	ShouldCompressLayers() bool
 
+	// SupportsForeignLayers returns true iff foreign layers in manifest should be actually
+	// uploaded to the image destination, false otherwise.
+	SupportsForeignLayers() bool
```

Comment:
This naming is pretty confusing; this should return `false` exactly in the cases when the destination _understands_ URLs.

As is, this is really `ShouldConvertForeignLayersToBlobCopies` :) Or perhaps keep the current semantics and call it `AcceptsForeignLayerURLs`?